relayer/chains/stacks/keys.go

```diff
@@ -0,0 +1,168 @@
+	"go.uber.org/zap"
+)
+
+func (p *Provider) RestoreKeystore(ctx context.Context) error {
+	keystorePath := p.keystorePath(p.cfg.Address)
+
+	encryptedKey, err := os.ReadFile(keystorePath)
+	if err != nil {
+		p.log.Error("Failed to read keystore file", zap.String("path", keystorePath), zap.Error(err))
+		return fmt.Errorf("failed to read keystore file: %w", err)
+	}
+
+	privateKey, err := p.kms.Decrypt(ctx, encryptedKey)
+	if err != nil {
+		p.log.Error("Failed to decrypt keystore", zap.Error(err))
+		return fmt.Errorf("failed to decrypt keystore: %w", err)
+	}
+
+	p.privateKey = privateKey
+
+	p.log.Info("Keystore restored successfully", zap.String("address", p.cfg.Address))
+	return nil
+}
+
+func (p *Provider) NewKeystore(passphrase string) (string, error) {
+	newKey, err := btcec.NewPrivateKey()
+	if err != nil {
+		p.log.Error("Failed to generate new private key", zap.Error(err))
+		return "", fmt.Errorf("failed to generate new private key: %w", err)
+	}
+	privateKeyBytes := append(newKey.Serialize(), byte(0x01))
+
+	network, err := MapNIDToChainID(p.cfg.NID)
+	if err != nil {
+		p.log.Error("Chain id not found. Check the NID config", zap.Error(err))
+		return "", fmt.Errorf("chain id not found: %w", err)
+	}
+
+	address, err := crypto.GetAddressFromPrivateKey(privateKeyBytes, network)
+	if err != nil {
+		p.log.Error("Failed to derive address from private key", zap.Error(err))
+		return "", fmt.Errorf("failed to derive address: %w", err)
+	}
+
+	encryptedKey, err := p.kms.Encrypt(context.Background(), privateKeyBytes)
+	if err != nil {
+		p.log.Error("Failed to encrypt private key", zap.Error(err))
+		return "", fmt.Errorf("failed to encrypt private key: %w", err)
+	}
+
+	encryptedPass, err := p.kms.Encrypt(context.Background(), []byte(passphrase))
+	if err != nil {
+		p.log.Error("Failed to encrypt passphrase", zap.Error(err))
+		return "", fmt.Errorf("failed to encrypt passphrase: %w", err)
+	}
+
+	keystorePath := p.keystorePath(address)
+	passPath := keystorePath + ".pass"
+
+	if err := os.MkdirAll(filepath.Dir(keystorePath), 0700); err != nil {
+		p.log.Error("Failed to create keystore directory", zap.Error(err))
+		return "", fmt.Errorf("failed to create keystore directory: %w", err)
+	}
+
+	if err := os.WriteFile(keystorePath, encryptedKey, 0600); err != nil {
+		p.log.Error("Failed to write keystore file", zap.String("path", keystorePath), zap.Error(err))
+		return "", fmt.Errorf("failed to write keystore file: %w", err)
+	}
+
+	if err := os.WriteFile(passPath, encryptedPass, 0600); err != nil {
+		p.log.Error("Failed to write passphrase file", zap.String("path", passPath), zap.Error(err))
+		return "", fmt.Errorf("failed to write passphrase file: %w", err)
+	}
+
+	p.privateKey = privateKeyBytes
+	p.cfg.Address = address
+
+	p.log.Info("New keystore created successfully", zap.String("address", address))
+	return address, nil
+}
+
+func (p *Provider) ImportKeystore(ctx context.Context, keyPath, passphrase string) (string, error) {
```

Comment:
The import keystore is normally used to import keys generated using external commands may be wallets into the relayer. So, the input may be file generated by cli commands or private key which would not be encrypted with the kms.